extensions/camel/camel-salesforce/deployment/src/main/java/io/quarkus/camel/component/salesforce/deployment/XStreamProcessor.java

```diff
@@ -0,0 +1,210 @@
+import com.thoughtworks.xstream.mapper.Mapper;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+
+public class XStreamProcessor {
+
+    private static final String[] INTERFACES_TO_REGISTER = {
+            Converter.class.getName(),
+            Mapper.class.getName()
+    };
+
+    private static final Set<String> EXCLUDED_CLASSES = new HashSet<>(Arrays.asList(CGLIBMapper.class.getName()));
+
+    @BuildStep(applicationArchiveMarkers = "com/thoughtworks/xstream/XStream.class")
+    void process(CombinedIndexBuildItem indexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer) {
+        for (String className : INTERFACES_TO_REGISTER) {
+            for (ClassInfo i : indexBuildItem.getIndex().getAllKnownImplementors(DotName.createSimple(className))) {
+                String name = i.name().toString();
+                if (!EXCLUDED_CLASSES.contains(name)) {
+                    reflectiveClassBuildItemBuildProducer.produce(new ReflectiveClassBuildItem(false, false, name));
+                }
+            }
+        }
+        reflectiveClassBuildItemBuildProducer.produce(new ReflectiveClassBuildItem(false, false,
+                Mapper.class,
+                Mapper.Null.class,
+                ConverterRegistry.class,
+                ConverterLookup.class,
+                ClassLoaderReference.class,
+                ReflectionProvider.class,
+                AnnotationConfiguration.class));
+
+        reflectiveClassBuildItemBuildProducer.produce(new ReflectiveClassBuildItem(false, false,
+                Boolean.class,
+                Byte.class,
+                Character.class,
+                Short.class,
+                Integer.class,
+                Long.class,
+                Float.class,
+                Double.class,
+                java.lang.Void.class,
+                java.lang.Object.class));
+
+        reflectiveClassBuildItemBuildProducer
+                .produce(new ReflectiveClassBuildItem(false, false, false, Class.class, ClassLoader.class));
+
+        reflectiveClassBuildItemBuildProducer.produce(new ReflectiveClassBuildItem(false, false,
+                "com.thoughtworks.xstream.core.JVM$Test",
+                "com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider",
+                "com.thoughtworks.xstream.core.JVM",
+                "com.thoughtworks.xstream.converters.reflection.SunUnsafeReflectionProvider",
+                "com.thoughtworks.xstream.converters.reflection.FieldUtil15",
+                "java.lang.Void",
+                "java.lang.Object",
+                "com.thoughtworks.xstream.core.util.CustomObjectOutputStream",
+                "java.awt.Color",
+                "javax.swing.LookAndFeel",
```

Comment:
Do we really want to include those two?

Reply:
We don't really have any choice at the moment, as it blows up without them. We may be able to do some substitutions later to deal with this but it will be fairly involved.